src/foam/nanos/jetty/HttpServer.js

```diff
@@ -125,12 +133,39 @@ foam.CLASS({
           }
         });
 
+        addJettyShutdownHook(server);
         server.setHandler(handler);
         server.start();
       } catch(Exception e) {
         e.printStackTrace();
       }
       `
+    },
+    {
+      name: 'addJettyShutdownHook',
```

Comment:
I feel like this could have just been inlined, but not a big deal, do what feels right for you in this case.